pkg/controller/accountclaim/reuse.go

```diff
@@ -185,23 +189,24 @@ func (r *ReconcileAccountClaim) cleanUpAwsAccount(reqLogger logr.Logger, awsClie
 		go cleanUpFunc(reqLogger, awsClient, awsNotifications, awsErrors) //nolint; cleanUpFunc // Not checking return value of goroutine
 	}
 
+	var err error
 	// Wait for clean up functions to end
 	for i := 0; i < len(cleanUpFunctions); i++ {
 		select {
 		case msg := <-awsNotifications:
 			reqLogger.Info(msg)
 		case errMsg := <-awsErrors:
-			err := errors.New(errMsg)
+			err = errors.New(errMsg)
 			reqLogger.Error(err, errMsg)
 			cleanUpStatusFailed = true
 		}
 	}
 
 	// Return an error if we saw any errors on the awsErrors channel so we can make the reused account as failed
 	if cleanUpStatusFailed {
-		cleanUpStatusFailedMsg := "Failed to clean up AWS account"
-		err := errors.New(cleanUpStatusFailedMsg)
+		cleanUpStatusFailedMsg := "failed to clean up AWS account"
 		reqLogger.Error(err, cleanUpStatusFailedMsg)
+		return err
```

Comment:
As a result of moving the err var creation to outside the loop on line 192, this returned err will be the last err we retrieved from the cleangoFunc which failed

Reply:
Adding the return err here is possibly the most important aspect of this PR - previously if we encountered an error while cleaning up AWS resources, we would still return nil and remove the finalizer

Reply:
Adding the return err here is possibly the most important aspect of this PR - previously if we encountered an error while cleaning up AWS resources, we would still return nil and remove the finalizer